lib/rex/socket/range_walker.rb

```diff
@@ -79,8 +79,21 @@ def parse(parseme)
     parseme.split(', ').map{ |a| a.split(' ') }.flatten.each do |arg|
       opts = {}
 
-      # Handle IPv6 first (support ranges, but not CIDR)
-      if arg.include?(":")
+      # Handle IPv6 CIDR first
+      if arg.include?(':') && arg.include?('/')
+        return false if !valid_cidr_chars(arg)
```

Comment:
`if !` vs. `unless` below.

Reply:
I think the way you have it might read better, at least for me.

Reply:
With these sorts of interrogators I feel `if !` reads better. If you come up with a name that reads well with `unless` I'l be happy to change it.

Reply:
Nope, we're in agreement.